cfpb/consumerfinance.gov
# Stub retirement API in Cypress tests

Similar to #7271, this change intercepts requests to the retirement API in Cypress tests and stubs a response. Since the retirement API hits ssa.gov, the Cypress tests will fail if ssa.gov is slow, down, or unreachable, and that's not something we want our integration tests to fail on.

---

## Additions

- Retirement API fixture

## Changes

- Intercept and stub retirement API in Cypress tests

## How to test this PR

1. Run the Before You Claim Cypress tests with `yarn cypress open` and selecting the `before-you-claim.cy.js` spec
2. Note the `(xhr) GET 200 /consumer-tools/retirement/retirement-api/estimator/1-1-1980/115000/` in step 10 shows that the response was stubbed (see screenshot below)

## Screenshots

Before | After
--- | ---
![before](https://user-images.githubusercontent.com/1862695/194406286-51702bd8-a97f-4777-9e60-f67505919c94.png) | ![after](https://user-images.githubusercontent.com/1862695/194406306-7351ee7a-9ebd-42ec-8325-e6533b518468.png)

## Notes

There are some values in the JSON that the real retirement API would calculate—like age based on the date of birth that was entered—that will no longer be updated in the fixture as time goes on. I don't think that will matter and the tests should keep passing fine, but I guess if they start failing after Jan. 1, that fixture is a good place to start looking.

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
